### ISSUE TITLE
NDK r10d GCC 4.9 x86 Compile

### ISSUE THREAD

**Issue by jimmaay** [xroche/coffeecatch#16]
Android ndk-build x86 compile returns an error: "redeclaration of struct 'struct _libc_fpreg'" which is already defined in "\ndk\platforms\android-9\arch-x86\usr\include\sys\ucontext.h". This patch fixed it with some small changes (https://gist.github.com/VladislavLipskiy/2a5014cfb48c0600df69). armv7a and arm seem to work fine.

**Comment by ghazel** [comment]
+1

**Comment by ghazel** [comment]
Personally I just did this:

```diff
diff --git a/coffeecatch.c b/coffeecatch.c
index 256a1a0..a8ce1e5 100644
--- a/coffeecatch.c
+++ b/coffeecatch.c
@@ -114,6 +114,7 @@ typedef struct ucontext {
 
 /* Taken from Google Breakpad. */
 
+#ifndef _SYS_UCONTEXT_H_
 /* 80-bit floating-point register */
 struct _libc_fpreg {
   unsigned short significand[4];
@@ -173,6 +174,8 @@ typedef struct ucontext {
 } ucontext_t;
 #endif
 
+#endif
+
 #elif defined(__mips__)
 
 /* Taken from Google Breakpad. */
```

But, same problem, similar effect.

**Comment by VioletGiraffe** [comment]
Yep, and there's a whole lot of redefined stuff from that header, not just one struct.

**Comment by gtbX** [comment]
Looks like someone else had to do something similar: https://github.com/xroche/coffeecatch/issues/25

**Comment by xroche** [comment]
This targets an NDK and Android version that are long out of support (NDK r10-era, Android 4-5, Dalvik) and it has not reproduced on a modern NDK. Closing as stale; please reopen with a repro on a current toolchain if it still happens.
